File: app/utils/storage/StrEnum.lua
local ____lualib = require("lualib_bundle")
local __TS__ArrayReduce = ____lualib.__TS__ArrayReduce
local ____exports = {}
function ____exports.newStrEnum(arr)
    return __TS__ArrayReduce(
        arr,
        function(____, acc, val)
            acc[val] = val
            return acc
        end,
        {}
    )
end
return ____exports

File: src/utils/storage/StrEnum.ts
/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-return */

export type StrEnum<T extends string> = { [K in T]: K };

export function newStrEnum<T extends string>(arr: readonly T[]): StrEnum<T> {
    return arr.reduce((acc, val) => {
        acc[val] = val;
        return acc;
    }, {} as any);
}